docs/docs/introducing-jsx.md

```diff
@@ -0,0 +1,206 @@
+---
+id: introducing-jsx
+title: Introducing JSX
+permalink: docs/introducing-jsx.html
+prev: hello-world.html
+next: rendering-elements.html
+---
+
+Consider this variable declaration:
+
+```js
+const element = <h1>Hello, world!</h1>;
+```
+
+This funny tag syntax is neither a string nor HTML.
+
+It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+
+>**Note:**
+>
+>Many people get turned off by JSX the first time they see it. One might think it violates separation of concerns but we believe [otherwise](https://www.youtube.com/watch?v=x7cQ3mrcKaY).
+>
+>We encourage you to [give it five minutes](https://signalvnoise.com/posts/3124-give-it-five-minutes). Facebook and many other companies successfully use it in production both for simple and complex apps.
+>
+>Also note that JSX is optional, and you can [use React without JSX](/react/docs/react-without-jsx.html).
+
+### Embedding Expressions in JSX
+
+JSX comes with the full power of JavaScript.
+
+You can embed any [JavaScript expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) in JSX by wrapping it in curly braces.
+
+For example, `2 + 2`, `user.name`, and `formatName(user)` are all valid expressions:
+
+```js{12}
+function formatName(user) {
+  return user.firstName + ' ' + user.lastName;
+}
+
+const user = {
+  firstName: 'Harper',
+  lastName: 'Perez'
+};
+
+const element = (
+  <h1>
+    Hello, {formatName(user)}!
+  </h1>
+);
+```
+
+**[Try it on Codepen.](http://codepen.io/gaearon/pen/PGEjdG?editors=0010)**
+
+Note that we wrapped JSX in parens and split it over multiple lines for readability.
+
+>**Caveat:**
```

Comment:
I think you could skip this caveat - when you're learning JSX this is going to be a distraction.